apps/wizard/pages/dashboard.py

```diff
@@ -0,0 +1,572 @@
+"""Create a dashboard with the main information about ETL steps, and the possibility to update them.
+
+"""
+import subprocess
+from datetime import datetime
+
+import streamlit as st
+from st_aggrid import AgGrid, GridUpdateMode, JsCode
+from st_aggrid.grid_options_builder import GridOptionsBuilder
+
+from apps.step_update.cli import StepUpdater
+from etl.config import ADMIN_HOST, ENV_IS_REMOTE
+from etl.db import can_connect
+
+# TODO:
+#  * Consider creating a script to regularly check for snapshot updates, fetch them and add them to the temporary DAG (this is the way that the "update state" will know if there are snapshot updates available).
+#  * Define a metric of update prioritization, based on number of charts (or views) and days to update. Sort steps table by this metric.
+
+# Current date.
+# This is used as the default version of new steps to be created.
+TODAY = datetime.now().strftime("%Y-%m-%d")
+
+# Define the base URL for the grapher datasets (which will be different depending on the environment).
+GRAPHER_DATASET_BASE_URL = f"{ADMIN_HOST}/admin/datasets/"
+if not GRAPHER_DATASET_BASE_URL.startswith(("http://", "https://")):
+    # Links in the steps table seem to only work if they start with "http://" or "https://".
+    # But ADMIN_HOST for staging servers starts with "staging-site-" and is not a valid URL.
+    # Therefore, prepend "http://" (not "https://" because the site is not secure, and the browser will block the link).
+    GRAPHER_DATASET_BASE_URL = f"http://{GRAPHER_DATASET_BASE_URL}"
+
+# List of dependencies to ignore when calculating the update state.
+# This is done to avoid a certain common dependency (e.g. population) to make all steps appear as needing major update.
+DEPENDENCIES_TO_IGNORE = [
+    # "data://garden/demography/2023-03-31/population",
+    "snapshot://hyde/2017/general_files.zip",
+]
+
+# Define labels for update states.
```

Comment:
Nit - you could also use Enum to make it a bit more readable.
```
from enum import Enum

class UpdateState(Enum):
    UNKNOWN = "Unknown"
    UP_TO_DATE = "No updates known"
    OUTDATED = "Outdated"
    MINOR_UPDATE = "Minor update possible"
    MAJOR_UPDATE = "Major update possible"
    ARCHIVABLE = "Archivable"

...
print(UpdateState.UNKNOWN)
```

Reply:
Thanks, nice idea! However, I just tried, and unfortunately it seems that streamlit is not happy about this kind of object:
```_pickle.PicklingError: Can't pickle <enum 'UpdateState'>: it's not the same object as __main__.UpdateState```
So I've reverted the changes.

Reply:
This could be solved by `enum.StrEnum`, which is avaialble since Python 3.11. Currently ETL supports >=py3.10, so not an option right now.

Instead, could use `UpdateState.UNKNOWN.value` instead of `UpdateState.UNKNOWN`. I've added these changes.